lib/org/chromium/webidl/BlinkIDLFileDAO.js

```diff
@@ -0,0 +1,160 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'BlinkIDLFileDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: 'DAO of Gitiles-backed web platform IDL files from Blink.',
+
+  requires: [
+    'foam.dao.InternalException',
+    'foam.dao.MDAO',
+    'foam.net.HTTPRequest',
+    'org.chromium.webidl.GitilesIDLFile',
+  ],
+
+  constants: {
+    branchToCommitURL: 'https://chromium.googlesource.com/chromium/src/+/master?format=JSON',
+    codeSearchQueryURL: 'https://cs.chromium.org/codesearch/json/search_request:1?search_request=b&query=file%3Athird_party%2FWebKit%2FSource.*idl%24+-file%3A%2Ftesting%2F+-file%3A%2Fbindings%2Ftests%2F&max_num_results=2000&internal_options=b&internal_options=e&search_request=e',
+    repositoryURL: 'https://chromium.googlesource.com/chromium/src.git',
+    gitilesBaseURL: 'https://chromium.googlesource.com/chromium/src/+',
+  },
+
+  properties: [
+    {
+      name: 'of',
+      value: 'org.chromium.webidl.GitilesIDLFile',
+    },
+    {
+      name: 'delegate',
+      factory: function() {
+        return this.MDAO.create({of: this.of});
+      },
+    },
+    {
+      class: 'String',
+      documentation: 'Git commit hash that DAO points to.',
+      name: 'commit',
+    },
+    {
+      class: 'Boolean',
+      documentation: `Internal state: Is the DAO ready to accept operations?
+        May throw a foam.dao.InternalException. To automatically catch retry
+        when not ready, wrap this DAO in a StoreAndForwardDAO.`,
+      name: 'isReady_',
+    },
+  ],
+
+  methods: [
+    function init() {
+      this.branchToCommit_().then(this.onCommitSet);
+    },
+    function put(obj) {
+      var notReady = this.checkNotReady_();
+      if (notReady) return notReady;
+
+      return this.delegate.put(obj);
```

Comment:
In all of these, you could write:

return this.checkNotReady_() || this.delegate.put(obj);